Engine/Settings.cs

```diff
@@ -73,7 +73,7 @@ public Settings(object settings, Func<string, string> presetResolver)
                 if (File.Exists(settingsFilePath))
                 {
                     filePath = settingsFilePath;
-                    parseSettingsFile(settingsFilePath);
+                    ParseSettingsFile(settingsFilePath);
```

Comment:
i'm wondering if we can simplify this constructor while we're here:
```c#
/// <summary>
/// Create a settings object from the input object.
/// </summary>
/// <param name="settings">An input object of type Hashtable or string.</param>
/// <param name="presetResolver">A function that takes in a preset and resolves it to a path.</param>
public Settings(object settings, Func<string, string> presetResolver)
{
    if (settings == null)
    {
        throw new ArgumentNullException(nameof(settings));
    }

    includeRules = new List<string>();
    excludeRules = new List<string>();
    severities = new List<string>();
    ruleArguments = new Dictionary<string, Dictionary<string, object>>(StringComparer.OrdinalIgnoreCase);

    // it can either be a preset or path to a file or a hashtable
    // handle the hashtable first
    var settingsHashtable = settings as Hashtable;
    if (settingsHashtable != null)
    {
        ParseSettingsHashtable(settingsHashtable);
        return;
    }

    // handle the present name or path
    string settingsFilePath = settings as string;
    if (settingsFilePath != null)
    {
        if (presetResolver != null)
        {
            var resolvedFilePath = presetResolver(settingsFilePath);
            if (resolvedFilePath != null)
            {
                settingsFilePath = resolvedFilePath;
            }
        }

        if (File.Exists(settingsFilePath))
        {
            filePath = settingsFilePath;
            ParseSettingsFile(settingsFilePath);
            return;
        }
    }

    throw new ArgumentException(Strings.SettingsInvalidType);
}
```

Reply:
Fixed by 417347a (with some additional simplification of `Settings.ParseSettingsFile`)